### ISSUE TITLE
list_buckets() and list_objects() are returning empty lists from pachyderm using minio client #4187

### ISSUE THREAD

**Issue by chaitanyak52** [pachyderm/pachyderm#4216]
Hi,
I am opening an existing issue 4187 as i could not see option to reopen after answer

@ysimonson, I have tried using list_buckets with minio library and works ok. I see a similar issue which is opened on minio-library and they have redirected to S3 incompatibility in the server implementation
Dear all,

when using list_buckets() or list_objects() on an account/bucket in pachyderm, the result is an empty list, allthough there are buckets/objects in that account.
```
s3_client = Minio(endpoint=s3_endpoint,
                  access_key=s3_accesskey,
                  secret_key=s3_secretkey,
                  region=s3_region,
                  secure=s3_secure)
the_bucket = s3_bucket
list_buckets = s3_client.list_buckets()  #Returns empty list
is_bucket = s3_client.bucket_exists(the_bucket)  #Returns True
list_objects = s3_client.list_objects(the_bucket)  #Returns empty generator object
[print(obj.object_name) for obj in list_objects]  #Throws the error below
```

**Environment?**:
- Kubernetes version (use `kubectl version`):
kubernetes version v12
- Pachyderm CLI and pachd server version (use `pachctl version`):
pachctl version         
COMPONENT           VERSION             
pachctl             1.9.2               
pachd               1.9.4  

- Cloud provider (e.g. aws, azure, gke) or local deployment 
custom kubernetes on azure 
- OS (e.g. from /etc/os-release):
Mac OS Sierra

Error on trying to access the objects from existing bucket

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/minio/parsers.py", line 104, in get_child_text
    return self.element.find('s3:{}'.format(name), _S3_NS).text
AttributeError: 'NoneType' object has no attribute 'text'

During handling of the above exception, another exception occurred:

```
Traceback (most recent call last):
  File "pachyderm_s3.py", line 75, in <module>
    main()
  File "pachyderm_s3.py", line 60, in main
    for obj in objects:
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/minio/api.py", line 906, in list_objects
    bucket_name=bucket_name)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/minio/parsers.py", line 238, in parse_list_objects
    is_truncated = root.get_child_text('IsTruncated').lower() == 'true'
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/minio/parsers.py", line 108, in get_child_text
    'Message: {}').format(self.root_name, name, error.message)
AttributeError: 'AttributeError' object has no attribute 'message'```

================================================

**Comment by pappasilenus** [comment]
@chaitanyak52 Can you link us to the minio client issue that references the S3 incompatibility?

**Comment by pappasilenus** [comment]
@chaitanyak52 checking in on this

**Comment by ysimonson** [comment]
Fixed in #4902, will go out with 1.11
